backend/api/communication/src/main/java/co/airy/core/api/communication/SendMessageController.java

```diff
@@ -78,22 +77,32 @@
         }
 
         final String userId = principalAccess.getUserId(auth);
+        final String channelId = Optional.ofNullable(channel).map((c) -> c.getId()).orElse("");
 
         final Message message = Message.newBuilder()
                 .setId(UUID.randomUUID().toString())
-                .setChannelId(channel.getId())
+                .setChannelId(channelId)
                 .setSourceRecipientId(payload.getSourceRecipientId())
                 .setContent(objectMapper.writeValueAsString(payload.getMessage()))
                 .setConversationId(conversationId)
                 .setHeaders(Map.of())
                 .setDeliveryState(DeliveryState.PENDING)
-                .setSource(channel.getSource())
+                .setSource(Optional.ofNullable(channel).map((c) -> c.getSource()).orElse(""))
                 .setSenderId(userId)
                 .setSentAt(Instant.now().toEpochMilli())
                 .setIsFromContact(false)
                 .build();
 
-        producer.send(new ProducerRecord<>(applicationCommunicationMessages.name(), message.getId(), message)).get();
-        return ResponseEntity.ok(MessageResponsePayload.fromMessageContainer(new MessageContainer(message, new MetadataMap())));
+        // If there is no channelId it implies that the converstaion was not found
+        // instally and it will be handled asynchronously
```

Comment:
```suggestion
        // If there is no channelId it implies that the conversation was not found
        // instantly and it will be handled asynchronously
```